File: atividade-16/atividade.py

class AFD:
    def __init__(self):
      self.state = 0

    def is_accepted(self, string):
      state = 0  
      for c in string:
          if state == 0:
              if c == '1':
                state = 0
              elif c == '0':
                state = 1
          elif state == 1:
            if c == '1':
              state = 0
            elif c == '0':
              state = 2
          elif state == 2:
              if c == '0':
                state = 2
              elif c == '1':
                pass
      return state == 2  

if __name__ == '__main__':
    afd = AFD()
    print(afd.is_accepted("100011"))   
    print(afd.is_accepted("10101"))   
    print(afd.is_accepted("000"))  
    print(afd.is_accepted("000110"))  
    print(afd.is_accepted("1"))    
    print(afd.is_accepted("100"))    

File: atividade-16/atividade.go
package main

import "fmt"

type AFD struct {
	state int
}

func NewAFD() *AFD {
	return &AFD{state: 0}
}

func (afd *AFD) IsAccepted(string string) bool {
	afd.state = 0
	for _, c := range string {
		if afd.state == 0 {
			if c == '1' {
				afd.state = 0
			} else if c == '0' {
				afd.state = 1
			}
		} else if afd.state == 1 {
			if c == '1' {
				afd.state = 0
			} else if c == '0' {
				afd.state = 2
			}
		} else if afd.state == 2 {
			if c == '0' {
				afd.state = 2
			} else if c == '1' {
				
			}
		}
	}
	return afd.state == 2
}

func main() {
	afd := NewAFD()
	fmt.Println(afd.IsAccepted("100011")) 
	fmt.Println(afd.IsAccepted("10101"))  
	fmt.Println(afd.IsAccepted("000"))    
	fmt.Println(afd.IsAccepted("000110")) 
	fmt.Println(afd.IsAccepted("1"))      
	fmt.Println(afd.IsAccepted("100"))    
}
